### ISSUE TITLE
`.condarc` Corruption Prevents Environment Initialization in Micromamba

### ISSUE THREAD

**Issue by ChaonengQuan** [mamba-org/mamba#3574]
### Troubleshooting docs

- [X] My problem is not solved in the Troubleshooting docs

### Anaconda default channels

- [X] I do NOT use the Anaconda default channels (pkgs/* etc.)

### How did you install Mamba?

Micromamba

### Search tried in issue tracker

Yes, but haven't found any similiar issue

### Latest version of Mamba

- [x] My problem is not solved with the latest version

### Tried in Conda?

I have this problem with Conda as well, without using Mamba

## Description

When a .condarc file in the home directory is corrupted, micromamba fails to initialize the Conda shell properly, preventing environments dependent on micromamba from loading. This issue impacts any application that relies on micromamba to activate environments and can leave users unable to access their workspace or tools when re-opening sessions.

## Impact
This issue affects open-source users by causing critical environments to fail when they rely on micromamba for activation. Any user with a corrupted .condarc file will be unable to start sessions dependent on the tool, resulting in failed load processes and inaccessible environments.

## Reproduction Steps
1. Create a new environment using micromamba.
2. Add a corrupted .condarc file in the home directory.
  
```bash
# Example
echo "random text string" > ~/.condarc
```

3. Attempt to initialize or re-open the environment.
4. Observe errors due to the initialization failure caused by the corrupted .condarc file.

## Proposed Solution
A suggested improvement would be to implement a safeguard or fallback mechanism within micromamba to bypass .condarc if it is unreadable or contains invalid configurations. For comparison, the .bashrc file in bash allows shell loading even if it contains invalid entries, enabling a more resilient startup process.

Thank you for considering this improvement to enhance micromamba's resilience against configuration file corruption.

### mamba info / micromamba info

```shell
libmamba version : 1.5.8
     micromamba version : 1.5.8
           curl version : libcurl/8.6.0 OpenSSL/3.2.1 zlib/1.2.13 zstd/1.5.5 libssh2/1.11.0 nghttp2/1.58.0
     libarchive version : libarchive 3.7.2 zlib/1.2.13 bz2lib/1.0.8 libzstd/1.5.5
       envs directories : /home/sagemaker-user/.conda/envs
                          /opt/conda/envs
          package cache : /home/sagemaker-user/.conda/pkgs
                          /opt/conda/pkgs
            environment : base (active)
           env location : /opt/conda
      user config files : /home/sagemaker-user/.mambarc
 populated config files : /opt/conda/.condarc
                          /opt/conda/.condarc
                          /etc/conda/.condarc
       virtual packages : __unix=0=0
                          __linux=5.10.226=0
                          __glibc=2.35=0
                          __archspec=1=x86_64-v4
               channels : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
       base environment : /opt/conda
               platform : linux-64
```

### Logs

```shell
eval "$(micromamba shell hook --shell=bash)"

micromamba activate base

Any `micromamba` command
```

**Comment by ChaonengQuan** [comment]
Fix for this bug, reviewed once with Julien (one of maintainers)
https://github.com/mamba-org/mamba/pull/3580

**Comment by JohanMabille** [comment]
Fixed in #3580 .

**Comment by jjerphan** [comment]
@ChaonengQuan this issue has been resolved in 1.5.11 (which is being released, see https://github.com/conda-forge/micromamba-feedstock/pull/226 and https://github.com/conda-forge/mamba-feedstock/pull/272) and in 2.0.4alpha3.

You can install 2.0.4alpha3 with:

```bash
micromamba install conda-forge/label/micromamba_prerelease::micromamba==2.0.4alpha3                                                                                                                                   
```
